src/test/java/edu/cornell/kfs/rass/batch/service/impl/RassSortServiceImplTest.java

```diff
@@ -0,0 +1,158 @@
+        LOG.info("testFullExtractAfterSort, failed: " + results.failedAgencies);
+        assertTrue("There shoulld be no agencies before their reports to agency after the sort", results.failedAgencies.size() == 0);
+    }
+
+    private void validateSortCount(int preSortCount, int postSortCount) {
+        assertEquals("The sort should return the same nnumber of agencies as was passed in", preSortCount, postSortCount);
+    }
+    
+    @Test
+    public void testNoParentAgengiesFile() throws JAXBException {
+        File xmlFile = new File(NO_PARENTS_FILE_NAME);
+        RassXmlDocumentWrapper rassXmlDocumentWrapper = cuMarshalService.unmarshalFile(xmlFile, RassXmlDocumentWrapper.class);
+        
+        int preSortCount = rassXmlDocumentWrapper.getAgencies().size();
+        List<RassXmlAgencyEntry> sortedAgencyEntries = rassSortServiceImpl.sortRassXmlAgencyEntriesForUpdate(rassXmlDocumentWrapper.getAgencies());
+        int postSortCount = sortedAgencyEntries.size();
+        
+        validateSortCount(preSortCount, postSortCount);
+    }
+    
+    @Test
+    public void testSingleAwardFile() throws JAXBException {
+        File xmlFile = new File(SINGLE_AWARD_FILE);
+        RassXmlDocumentWrapper rassXmlDocumentWrapper = cuMarshalService.unmarshalFile(xmlFile, RassXmlDocumentWrapper.class);
+        
+        int preSortCount = rassXmlDocumentWrapper.getAgencies().size();
+        List<RassXmlAgencyEntry> sortedAgencyEntries = rassSortServiceImpl.sortRassXmlAgencyEntriesForUpdate(rassXmlDocumentWrapper.getAgencies());
+        int postSortCount = sortedAgencyEntries.size();
+        
+        validateSortCount(preSortCount, postSortCount);
+    }
+    
+    @Test
+    public void testLeftOverAgencySort() throws JAXBException {
+        File xmlFile = new File(LEFT_OVER_AGENCY_TEST_FILE_NAME);
+        RassXmlDocumentWrapper rassXmlDocumentWrapper = cuMarshalService.unmarshalFile(xmlFile, RassXmlDocumentWrapper.class);
+        
+        int preSortCount = rassXmlDocumentWrapper.getAgencies().size();
+        List<RassXmlAgencyEntry> sortedAgencyEntries = rassSortServiceImpl.sortRassXmlAgencyEntriesForUpdate(rassXmlDocumentWrapper.getAgencies());
+        int postSortCount = sortedAgencyEntries.size();
+        
+        validateSortCount(preSortCount, postSortCount);
+        
+        String[] expectedAgencyNumbers = {"Cappa", "Alpha", "Delta", "Beta"};
+        
+        assertEquals("expected count should match actualCount", expectedAgencyNumbers.length, sortedAgencyEntries.size());
+        
+        int i = 0;
+        for (RassXmlAgencyEntry entry : sortedAgencyEntries) {
+            assertEquals("Agency index " + i + " should equal expected value", expectedAgencyNumbers[i], entry.getNumber());
+            i++;
+        }
+        
+    }
+    
+    private TestResults checkReportsToAgencyBeforeChildAgencies(List<RassXmlAgencyEntry> agencies) {
```

Comment:
this method is not really checking something but building a couple of lists. Please update the name to better describe what this method is doing.